pkg/util/cluster/cluster.go

```diff
@@ -447,6 +447,7 @@ func (c *Cluster) createCluster(ctx context.Context, vnetResourceGroup, clusterN
 				ResourceGroupID:      fmt.Sprintf("/subscriptions/%s/resourceGroups/%s", c.env.SubscriptionID(), "aro-"+clusterName),
 				FipsValidatedModules: api.FipsValidatedModulesEnabled,
 				Version:              osClusterVersion,
+				PullSecret:           api.SecureString(os.Getenv("USER_PULL_SECRET")),
```

Comment:
Do you want to validate if Pull Secret content is valid? I'm still okay with the approach here.

Reply:
I think we don't have to check it because this function is only used  for hack cluster creation.
And PULL_SECRET is set in the secret env file, so as long as we run `. ./env`, we won't set it explicitly.

Reply:
Makes sense